src/main/java/seedu/address/model/Model.java

```diff
@@ -65,48 +75,86 @@
      * Sets the user prefs' article data directory path.
      */
     void setArticleDataDirectoryPath(Path articleDataDirectoryPath);
+
+    /**
+     * Returns true if an entry with the same identity as {@code entry} exists in either the list/archives entry book.
+     */
+    boolean hasEntry(Entry entry);
```

Comment:
This is very well-done!

What I tried:
1. Add link
2. Archive link added in step 1
3. try to add same link

and got a "duplicate entry" error as expected.

Two related suggestions:
* assert this behaviour with a test
* make the error message more descriptive (this might not be easy)

Reply:
Thank you! I thought about this too in this comment https://github.com/CS2103-AY1819S2-W10-1/main/pull/106#commitcomment-32903533

For now this PR will not include tests, but I'll keep it in mind when I do them!